src/main/osd/osd_elements.c

```diff
@@ -1374,11 +1417,15 @@ void osdAnalyzeActiveElements(void)
         osdAddActiveElement(OSD_FLIGHT_DIST);
     }
 #endif // GPS
-
-#ifdef USE_ESC_SENSOR
     if (featureIsEnabled(FEATURE_ESC_SENSOR)) {
+        osdAddActiveElement(OSD_ESC_RPM);
+        osdAddActiveElement(OSD_ESC_RPM_FREQ);
         osdAddActiveElement(OSD_ESC_TMP);
+    }
+#ifdef USE_RPM_FILTER
+    if ((!featureIsEnabled(FEATURE_ESC_SENSOR)) && (motorConfig()->dev.useDshotTelemetry)) {
         osdAddActiveElement(OSD_ESC_RPM);
+        osdAddActiveElement(OSD_ESC_RPM_FREQ);
     }
 #endif
```

Comment:
This section needs to be revised. Elements can't be added twice. There is no "uniqueness" logic in `osdAddActiveElements()` - it just appends to a list. As it stands now the elements will be rendered twice every OSD update causing unnecessary processing.

Reply:
Sorry for my lack of understanding,  my understanding is  OSD_ESC_RPM_FREQ and OSD_ESC_RPM are two different elements. 

Inside osdAnalyzeActiveElements,  there are two places where they are added twice but they are mutual exclusive due to if condition imposed.

Reply:
The problem is that the first occurrence is only bounded by `if (featureIsEnabled(FEATURE_ESC_SENSOR))` and does not consider `#ifdef USE_ESC_SENSOR`.

This would probably be better for the entire section:
```
#ifdef USE_ESC_SENSOR
    if (featureIsEnabled(FEATURE_ESC_SENSOR)) {
        osdAddActiveElement(OSD_ESC_TMP);
    }
#endif

#if defined(USE_ESC_SENSOR) || defined(USE_RPM_FILTER)
    if (featureIsEnabled(FEATURE_ESC_SENSOR) || motorConfig()->dev.useDshotTelemetry) {
        osdAddActiveElement(OSD_ESC_RPM);
        osdAddActiveElement(OSD_ESC_RPM_FREQ);
    }
#endif
```